backend/routes.js

```diff
@@ -17,44 +17,67 @@ import {
 import fetch from 'node-fetch';
 
+const opn = require('opn');
 const HOSTS = '/etc/hosts';
 
-
-// return HTTPS certs if they exist for a server to use when created or null
-export function getCerts() {
-    try {
-        const keyFile = `${__dirname}/..${getSetting('KEY_FILE')}`;
-        const certFile = `${__dirname}/..${getSetting('CSR_FILE')}`;
-        return {
-            key: fs.readFileSync(keyFile),
-            certificate: fs.readFileSync(certFile)
-        };
-    } catch (e) {
-        return {};
-    }
-}
-
 export default class Server {
     constructor(args = {}) {
         this.certs = getCerts();
         /*
-        // if server is given a protocol argument, enforce it,
-        // otherwise decide which protocol to use
-        // depending if there are certificates
+         * If server is given a protocol argument, enforce it,
+         * otherwise decide which protocol to use
+         * depending if there are certificates. This is mainly
+         * used in tests.
         */
         const apiVersion = '1.0.0';
+        this.apiVersion = apiVersion;
         if (args.protocol === 'HTTP' || isEmpty(this.certs)) {
             this.server = restify.createServer({version: apiVersion});
             this.protocol = 'http';
             this.domain = 'localhost';
+            const redirectBaseUrl = `${this.protocol}://${this.domain}:${getSetting('PORT')}`;
+            const oauthClientId = 'isFcew9naom2f1khSiMeAtzuOvHXHuLwhPsM7oPt';
+            const oauthUrl = (
+                'https://plot.ly/o/authorize/?response_type=token&' +
+                `client_id=${oauthClientId}&` +
+                `redirect_uri=${redirectBaseUrl}/oauth2/callback`
+            );
+            // We wanna skip fetching certs in cases such as tests.
+            if (!args.skipFetchCerts) {
+                const users = getSetting('USERS');
+                /*
+                 * If no USERS in settings we need to go through OAuth before
+                 * being able to ask for a certificate from the CA.
+                 * Let the app load for a while before launching OAuth
+                 * to avoid opening the new tab in the browser without
+                 * the attention of the user.
+                 */
+                if (isEmpty(users)) {
+                    // `opn` module opens a tab in the default browser.
+                    setTimeout(() => opn(oauthUrl), 10000);
+                }
+                const checkOauthComplete = setInterval(() => {
+                    // TODO: As metnioned in ./certificates, not sure what
+                    // the way to go for multiple users should be.
+                    if (users[0]) {
+                        if (has('username', users[0]) && has('accessToken', users[0])) {
+                            fetchAndSaveCerts();
+                            clearInterval(checkOauthComplete);
+                        }
+                    }
+                }, 1000);
+            }
```

Comment:
this is a lot of logic to go into the `Server` constructor. It's not clear from the outsider that instantiating a `Server` would end up having so many side affects. Can we place this logic anywhere else?